.github/workflows/update-caches.yml

```
@@ -0,0 +1,136 @@
+name: Update Actions Caches
+
+on:
+  # Manually
+  workflow_dispatch:
+  # On PR merge
+  push:
+    branches:
+      - main
+  # After nightly release
+  schedule:
+    - cron: "0 1 * * *"
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_PROFILE_TEST_DEBUG: 0
+  CARGO_PROFILE_DEV_DEBUG: 0
+  # If nightly is breaking CI, modify this variable to target a specific nightly version.
+  NIGHTLY_TOOLCHAIN: nightly
+
+jobs:
+  env:
+    runs-on: ubuntu-latest
+    outputs:
+      NIGHTLY_TOOLCHAIN: ${{ steps.env.outputs.NIGHTLY_TOOLCHAIN }}
+      MSRV: ${{ steps.env.outputs.MSRV }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - name: get MSRV
+        id: msrv
+        run: |
+          msrv=`cargo metadata --no-deps --format-version 1 | jq --raw-output '.packages[] | select(.name=="bevy") | .rust_version'`
+          echo "MSRV=$msrv" >> $GITHUB_OUTPUT
+      - name: Expose Env
+        id: env
+        run: |
+          echo "NIGHTLY_TOOLCHAIN=${{ env.NIGHTLY_TOOLCHAIN }}" >> $GITHUB_OUTPUT
+
+  build-caches:
+    name: Build Caches
+    needs: ["env"]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            toolchain: stable
+            target: ""
+          - os: macos-latest
+            toolchain: stable
+            target: ""
+          - os: windows-latest
+            toolchain: stable
+            target: ""
+          - os: ubuntu-latest
+            toolchain: ${{ needs.env.outputs.NIGHTLY_TOOLCHAIN }}
+            target: ""
+          - os: ubuntu-latest
+            toolchain: ${{ needs.env.outputs.MSRV }}
+            target: ""
+          - os: macos-latest
+            toolchain: ${{ needs.env.outputs.NIGHTLY_TOOLCHAIN }}
+            target: ""
+          - os: ubuntu-latest
+            toolchain: ${{ needs.env.outputs.NIGHTLY_TOOLCHAIN }}
+            target: wasm32-unknown-unknown
+          - os: ubuntu-latest
+            toolchain: stable
+            target: wasm32-unknown-unknown
+          - os: ubuntu-latest
+            toolchain: stable
+            target: x86_64-unknown-none
+          - os: ubuntu-latest
+            toolchain: stable
+            target: thumbv6m-none-eabi
+          - os: ubuntu-latest
+            toolchain: stable
+            target: aarch64-linux-android
+          - os: macos-latest
+            toolchain: stable
+            target: aarch64-apple-ios-sim
+
+    steps:
+      - name: Get Date
+        id: get-date
+        run: |
+          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: Checkout Bevy main branch
+        uses: actions/checkout@v4
+        with:
+          repository: "bevyengine/bevy"
```

```suggestion
        with:
          repository: "bevyengine/bevy"
          ref: "main"
```

(Nit) Just to be safe :)